src/Polly.Core/Hedging/Controller/TaskExecution.cs

```diff
@@ -99,34 +99,43 @@ public async ValueTask<bool> InitializeAsync<TState>(
         _activeContext = _cachedContext;
         _activeContext.InitializeFrom(primaryContext, _cancellationSource!.Token);
 
-        if (primaryContext.CancellationToken.CanBeCanceled)
-        {
-            _cancellationRegistration = primaryContext.CancellationToken.Register(o => ((CancellationTokenSource)o!).Cancel(), _cancellationSource);
-        }
+#if NET
+        _cancellationRegistration = primaryContext.CancellationToken.UnsafeRegister(static o => ((CancellationTokenSource)o!).Cancel(), _cancellationSource);
```

Comment:
Why the `Unsafe` version, and what makes it safe to use?

Reply:
The unsafe version doesn't capture any async state for the registration. It's safe to use because the callback doesn't do anything other than call `CancellationTokenSource.Cancel()` (any registrations on that 2nd CTS would have their own async state, if they need it).